pandas/core/frame.py

```diff
@@ -4714,14 +4715,15 @@ def f(vals):
 
     @Substitution(**_shared_doc_kwargs)
     @Appender(NDFrame.sort_values.__doc__)
-    def sort_values(
+    def sort_values(  # type: ignore
```

Comment:
Can you post the error(s) these were causing? We typically try to avoid ignore

Reply:
The error is `pandas/core/frame.py:4718: error: Signature of "sort_values" incompatible with supertype "NDFrame"`. The problem seems to already exist in master, namely that `sort_values` in `core/generic.py` has `by=None` as a default argument, while it has no default in `core/frame.py`. This doesn't seem to relate to my changes. Similar problem with `sort_index` (namely, the `by` kwarg changing order). It seems like adding the `key` argument sets mypy off for some reason.

Reply:
@WillAyd what do you think? The problem is that `sort_values` in `generic.py` uses

```python
 4149     def sort_values(
 4150         self,
 4151         by=None,
 4152         axis=0,
 4153         ascending=True,
 4154         inplace=False,
 4155         kind="quicksort",
 4156         na_position="last"
 4157     ):
```

and in `core/frame.py` it is

```python
4718     def sort_values(
4719         self,
4720         by,
4721         axis=0,
4722         ascending=True,
4723         inplace=False,
4724         kind="quicksort",
4725         na_position="last",
4726         key: Optional[Callable] = None
4727     ):
```

so `by` has a default in the base class, but no default in the child. There also isn't ven a `by` keyword in `core/series.py`. This isn't related to a change I made. It just seems to be a kind of unfortunate inheritance issue related to the different needs of `series` and `frame`.

Reply:
Does fixing it in the base class resolve? I think preferable to all of the type ignores if that is all that is required

Reply:
@WillAyd I don't think we can fix it by changing the base class. The `generic.py` `sort_values` has `by` as the first argument, while the `series.py` `sort_values` doesn't even have `by` as an argument (for obvious reasons). `frame.py` has it has a first argument. I don't see any change that can fix it – the APIs are just fundamentally different.

Reply:
The one fix would just be to make `sort_values` in the NDFrame base class use `def sort_values(*args, **kwargs)`, and let child classes redefine it.

Reply:
@WillAyd I was able to make some changes to the child classes that fixed the type errors without any major changes. Type annotations are a little less verbose, but there are no `type:ignore` calls anywhere.